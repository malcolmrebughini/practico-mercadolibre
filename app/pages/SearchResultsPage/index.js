import { connect } from 'react-redux';
import actions from '../../actions';
import SearchResultsPage from './SearchResultsPage';


function mapStateToProps(state, ownProps) {
  return {
    itemsCount: (state.search.items || []).length,
    isLoading: state.search.isLoading,
    q: ownProps.location.search.replace('?q=', ''),
  };
}


export default connect(mapStateToProps, actions)(SearchResultsPage);
