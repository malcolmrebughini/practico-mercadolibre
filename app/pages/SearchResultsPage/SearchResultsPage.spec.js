/* eslint-env jasmine */
/* eslint react/jsx-filename-extension: 0 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchResultsPage from './SearchResultsPage';


describe('SearchResultsPage', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <SearchResultsPage
        q="Mala Fama"
        isLoading={false}
        search={jest.fn()}
        updateInput={jest.fn()}
        itemsCount={4}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call search and updateInput if no items', () => {
    const searchCallback = jest.fn();
    const updateInputCallback = jest.fn();

    shallow(
      <SearchResultsPage
        search={searchCallback}
        updateInput={updateInputCallback}
        itemsCount={0}
        q="Mala Fama"
      />,
    );

    expect(searchCallback.mock.calls.length).toEqual(1);
    expect(searchCallback.mock.calls[0]).toEqual(['Mala Fama']);
    expect(updateInputCallback.mock.calls.length).toEqual(1);
    expect(updateInputCallback.mock.calls[0]).toEqual(['Mala Fama']);
  });

  it('should call clearData on unmount', () => {
    const callback = jest.fn();
    shallow(
      <SearchResultsPage
        search={jest.fn()}
        updateInput={jest.fn()}
        itemsCount={0}
        q="Mala Fama"
        clearData={callback}
      />,
    ).unmount();

    expect(callback.mock.calls.length).toEqual(1);
  });
});
