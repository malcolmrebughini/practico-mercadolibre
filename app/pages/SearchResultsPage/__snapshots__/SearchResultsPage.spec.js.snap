// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsPage should render correctly 1`] = `
<div>
  <Connect(TopBar) />
  <div>
    <HelmetWrapper
      encodeSpecialCharacters={true}
    >
      <title>
        Mala Fama
         en MercadoLibre
      </title>
      <meta
        content="Encontrá Mala Fama en Mercado Libre. Descubrí la mejor forma de comprar online."
        name="description"
      />
    </HelmetWrapper>
    <Connect(Breadcrumbs) />
    <Connect(ItemsList) />
  </div>
</div>
`;
